src/headers.rs

```diff
@@ -139,21 +139,30 @@ impl TryFrom<&[u8]> for Headers {
                 return parse_error("version line does not begin with NATS/1.0");
             }
 
-            if line.len() > HEADER_LINE_LEN {
-                let status_line = &line[HEADER_LINE_LEN..];
-                let mut parts = status_line.split_whitespace();
-
-                if let Some(status) = parts.next() {
-                    let entry = inner
-                        .entry(STATUS_HEADER.to_string())
-                        .or_insert_with(HashSet::default);
-                    entry.insert(status.to_string());
-
-                    if let Some(description) = parts.next() {
-                        let entry = inner
-                            .entry(DESCRIPTION_HEADER.to_string())
-                            .or_insert_with(HashSet::default);
-                        entry.insert(description.to_string());
+            if let Some(slice) = line.get(HEADER_LINE_LEN..).map(|s| s.trim()) {
+                match slice.split_once(' ') {
```

Comment:
What if 2 spaces between code and description?

Reply:
Got a test case for that, splits on first space then trims 🙂